include/deal.II/lac/precondition.h

```diff
@@ -2451,44 +2451,52 @@ namespace internal
         internal::set_initial_guess(temp_vector1);
         data.constraints.set_zero(temp_vector1);
 
-        if (data.eigenvalue_algorithm == internal::EigenvalueAlgorithm::lanczos)
+        // note: temp_vector1 only contains zeroes, which happens if all dofs
+        // are constrained, there is nothing to do
+        if (temp_vector1.l2_norm() != 0.0)
```

Comment:
Is it possible to use `all_zero()` here? We do require that for the vector space vector concept (which this template doesn't check for).

Reply:
Indeed, that should be cheaper.